library/ssl_cli.c

```diff
@@ -312,24 +332,22 @@ static void ssl_write_supported_elliptic_curves_ext( mbedtls_ssl_context *ssl,
     *p++ = (unsigned char)( ( ( elliptic_curve_len     )      ) & 0xFF );
 
     *olen = 6 + elliptic_curve_len;
+
+    return( 0 );
 }
 
-static void ssl_write_supported_point_formats_ext( mbedtls_ssl_context *ssl,
-                                                   unsigned char *buf,
-                                                   size_t *olen )
+static int ssl_write_supported_point_formats_ext( mbedtls_ssl_context *ssl,
+                                                  unsigned char *buf,
+                                                  const unsigned char *end,
+                                                  size_t *olen )
 {
     unsigned char *p = buf;
-    const unsigned char *end = ssl->out_msg + MBEDTLS_SSL_MAX_CONTENT_LEN;
+    (void) ssl; /* ssl used for debugging only */
```

Comment:
Can we wrap this with an `#ifdef` for debugging instead of always casting to `(void)`?

Reply:
@Patater Violates DRY by repeating the condition under which the debug macros use `ssl`.

Arguably the debug macros should be defined as `((void)(ssl))` when debugging is off.

Reply:
That'd make the debug macros easier to use without warnings.